data/src/main/java/com/linkedin/data/schema/annotation/PegasusExtensionSchemaAnnotationHandler.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright (c) 2020 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.data.schema.annotation;
+
+import com.linkedin.data.DataMap;
+import com.linkedin.data.schema.PathSpec;
+import com.linkedin.data.schema.compatibility.CompatibilityMessage;
+import java.util.List;
+import java.util.Map;
+import org.apache.commons.lang3.tuple.Pair;
+
+
+/**
+ * This SchemaAnnotationHandler is used to check extension schema annotation(@extension) compatibility
+ *
+ * @author Yingjie Bi
+ */
+public class PegasusExtensionSchemaAnnotationHandler implements SchemaAnnotationHandler
+{
+  private final String EXTENSION_ANNOTATION_NAMESPACE = "extension";
+
+  @Override
+  public ResolutionResult resolve(List<Pair<String, Object>> propertiesOverrides,
+      ResolutionMetaData resolutionMetadata)
+  {
+    return new ResolutionResult();
+  }
+
+  @Override
+  public String getAnnotationNamespace()
+  {
+    return EXTENSION_ANNOTATION_NAMESPACE;
+  }
+
+  @Override
+  public AnnotationValidationResult validate(Map<String, Object> resolvedProperties, ValidationMetaData metaData)
+  {
+    return new AnnotationValidationResult();
+  }
+
+  @Override
+  public SchemaVisitor getVisitor()
+  {
+    // No need to override properties, use IdentitySchemaVisitor to skip schema traverse.
+    return new IdentitySchemaVisitor();
+  }
+
+  @Override
+  public boolean implementsCheckCompatibility()
+  {
+    return true;
+  }
+
+  @Override
+  public AnnotationCompatibilityResult checkCompatibility(Map<String, Object> prevResolvedProperties, Map<String, Object> currResolvedProperties,
```

Comment:
Wonder if we can change `prevResolvedProperties` to `snapshot...properties`, because it is actually from the snapshot.

Reply:
we have converted snapshot.pdl to DataSchema for compatibility check. Technically it is dataschema, not snapshot. So calling it 'snapshot...' is not accurate.